Sources/OpenAPIRuntime/Conversion/Converter+Client.swift

```diff
@@ -142,53 +110,42 @@ extension Converter {
 
     //    | client | set | request body | binary | optional | setOptionalRequestBodyAsBinary |
     public func setOptionalRequestBodyAsBinary(
-        _ value: Data?,
-        headerFields: inout [HeaderField],
+        _ value: HTTPBody?,
+        headerFields: inout HTTPFields,
         contentType: String
-    ) throws -> Data? {
+    ) throws -> HTTPBody? {
         try setOptionalRequestBody(
             value,
             headerFields: &headerFields,
             contentType: contentType,
-            convert: convertDataToBinary
+            convert: { $0 }
         )
     }
 
     //    | client | set | request body | binary | required | setRequiredRequestBodyAsBinary |
     public func setRequiredRequestBodyAsBinary(
-        _ value: Data,
-        headerFields: inout [HeaderField],
+        _ value: HTTPBody,
+        headerFields: inout HTTPFields,
         contentType: String
-    ) throws -> Data {
+    ) throws -> HTTPBody {
         try setRequiredRequestBody(
             value,
             headerFields: &headerFields,
             contentType: contentType,
-            convert: convertDataToBinary
-        )
-    }
-
-    //    | client | get | response body | string | required | getResponseBodyAsString |
-    public func getResponseBodyAsString<T: Decodable, C>(
-        _ type: T.Type,
-        from data: Data,
-        transforming transform: (T) -> C
-    ) throws -> C {
-        try getResponseBody(
-            type,
-            from: data,
-            transforming: transform,
-            convert: convertFromStringData
+            convert: { $0 }
         )
     }
 
     //    | client | get | response body | JSON | required | getResponseBodyAsJSON |
     public func getResponseBodyAsJSON<T: Decodable, C>(
         _ type: T.Type,
-        from data: Data,
+        from data: HTTPBody?,
         transforming transform: (T) -> C
-    ) throws -> C {
-        try getResponseBody(
+    ) async throws -> C {
+        guard let data else {
+            throw RuntimeError.missingRequiredResponseBody
+        }
```

Comment:
Because we used to return empty data even for responses with no documented payload, it was the discussions we had about empty vs nil bodies. We ended up making it all consistent in SOAR-0004/5, and now both request and response bodies in the transport and middleware protocols are optional, so these converter helper functions need to take them as-are, and they unwrap the values at runtime.